src/apps/concepts/__test__/components/ConceptsTable.test.tsx

```diff
@@ -0,0 +1,145 @@
+import React from "react";
+import { render } from "../../../../test-utils";
+import "@testing-library/jest-dom";
+import ConceptsTable from "../../components/ConceptsTable";
+import {
+  ConceptName,
+  ConceptDescription,
+  APIConcept,
+  APIMapping,
+} from "../../types";
+
+type conceptsTableProps = React.ComponentProps<typeof ConceptsTable>;
+const baseProps: conceptsTableProps = {
+  concepts: [],
+  buttons: {},
+  q: "",
+  setQ: function() {},
+  page: 1,
+  sortDirection: "sortAsc",
+  sortBy: "id",
+  limit: 10,
+  buildUrl: function() {},
+  goTo: function() {},
+  count: 5,
+  toggleShowOptions: function() {},
+  addConceptsToDictionary: function() {},
+  linkedDictionary: "",
+  linkedSource: "",
+  canModifyConcept: () => {
+    return true;
+  },
+  removeConceptsFromDictionary: function() {},
+};
+
+const testConceptName: ConceptName = {
+  name: "test concept",
+  locale: "en",
+  external_id: "1234",
+  locale_preferred: true,
+  name_type: "Fully Specified",
+};
+
+const testDescription: ConceptDescription = {
+  description: "concept description",
+  locale: "en",
+  external_id: "3456",
+  locale_preferred: true,
+};
+
+const testMapping: APIMapping = {
+  map_type: "test",
+  external_id: "79787",
+  from_concept_url: "from_concept_url",
+  url: "concept_url",
+  retired: false,
+  to_concept_code: "concept_code",
+};
+const testConcept: APIConcept = {
+  id: "123",
+  external_id: "234234",
+  concept_class: "Diagnosis",
+  datatype: "Numeric",
+  names: [testConceptName],
+  descriptions: [testDescription],
+  url: "abcd",
+  version_url: "version_url",
+  extras: {},
+  display_name: "display_name",
+  mappings: [testMapping],
+  retired: false,
+  source_url: "source_url",
+};
+
+function renderUI(props: Partial<conceptsTableProps> = {}) {
+  return render(<ConceptsTable {...baseProps} {...props} />);
+}
+
+describe("Action Icon for the Concepts in Concepts Table", () => {
```

Comment:
@rakesh-1010 @kirity-mahanty can we also have a snapshot test for this ??

Reply:
This component is having the `TablePagination` component. Each time it creates randomly generated id. So snapshot test is not possible for this component.
https://github.com/mui-org/material-ui/issues/21293